### ISSUE TITLE
Composite Foreign Keys and Schema reflection

### ISSUE THREAD

**Issue by HavokInspiration** [cakephp/cakephp#6789]
Take this SQL queries (from the [MySQL official documentation](https://dev.mysql.com/doc/refman/5.5/en/create-table-foreign-keys.html)) :

```sql
CREATE TABLE product (
    category INT NOT NULL, id INT NOT NULL,
    price DECIMAL,
    PRIMARY KEY(category, id)
)   ENGINE=INNODB;

CREATE TABLE customer (
    id INT NOT NULL,
    PRIMARY KEY (id)
)   ENGINE=INNODB;

CREATE TABLE product_order (
    no INT NOT NULL AUTO_INCREMENT,
    product_category INT NOT NULL,
    product_id INT NOT NULL,
    customer_id INT NOT NULL,

    PRIMARY KEY(no),
    INDEX (product_category, product_id),
    INDEX (customer_id),

    FOREIGN KEY (product_category, product_id)
      REFERENCES product(category, id)
      ON UPDATE CASCADE ON DELETE RESTRICT,

    FOREIGN KEY (customer_id)
      REFERENCES customer(id)
)   ENGINE=INNODB;
```

Notice that the ``product_order`` table has a foreign key that has references to both the ``category`` and ``id`` columns of the ``product`` table.

If you do : 

```php
$table = TableRegistry::get('ProductOrder');
debug($table->schema());
```

This output :

```php
object(Cake\Database\Schema\Table) {
	[protected] _table => 'product_order'
 //[...]
	[protected] _indexes => [
		'product_category' => [
			'type' => 'index',
			'columns' => [
				(int) 0 => 'product_category',
				(int) 1 => 'product_id'
			],
			'length' => []
		],
		'customer_id' => [
			'type' => 'index',
			'columns' => [
				(int) 0 => 'customer_id'
			],
			'length' => []
		]
	]
	[protected] _constraints => [
 //[...]
		'product_order_ibfk_1' => [
			'type' => 'foreign',
			'columns' => [
				(int) 0 => 'product_id'
			],
			'references' => [
				(int) 0 => 'product',
				(int) 1 => 'id'
			],
			'update' => 'cascade',
			'delete' => 'restrict',
			'length' => []
		],
        //[...]
	]
//[...]
}
```

As you can see, the indexes are correct, both the column are referenced.
However, the constraints only shows the ``product_id`` column. It seems to come from the fact that when a constraint is added, it replaces any already defined constraint having the same name [(source)](https://github.com/cakephp/cakephp/blob/master/src/Database/Schema/Table.php#L517) 

The value for ``product_order_ibfk_1`` should probably be :

```php
'type' => 'foreign',
'columns' => [
    'product_id',
    'product_category'
],
'references' => [
    [
        'product',
        'id'
    ],
    [
        'product',
        'category'
    ]
],
'update' => 'cascade',
'delete' => 'restrict',
'length' => []
```
Note that I only tested this for MySQL and if you do a :

```php
$table->schema()->createSql(ConnectionManager::get('default'));
```

It outputs 

```sql
[...]
CONSTRAINT `product_order_ibfk_1` FOREIGN KEY (`product_id`) REFERENCES `product` (`id`) ON UPDATE CASCADE ON DELETE RESTRICT,
[...]
```

**Comment by HavokInspiration** [comment]
I started working on this.
I fixed SQLite and MySQL but I'm having difficulties with Postgres. The foreign keys reflection SQL request for Postgres gives bad results when multiple references are made.

![screenshot from 2015-06-14 16 22 28](https://cloud.githubusercontent.com/assets/5243386/8148915/d0093762-12b1-11e5-9cd4-522e2ae8b508.png)

This is apparently expected from what I gathered online.
I'm currently looking for a way to fix the request to have multiple columns properly returned.
